PYME/LMVis/visCore.py

```diff
@@ -416,10 +416,16 @@ def add_pointcloud_layer(self, method='points', ds_name='output', **kwargs):
         logger.debug('Added layer, datasouce=%s' % l.dsname)
         return l
 
-    def add_mesh_layer(self, method='shaded', ds_name='mesh0', **kwargs):
+    def add_mesh_layer(self, method='shaded', ds_name=None, **kwargs):
         from PYME.LMVis.layers.mesh import TriangleRenderLayer
         from PYME.misc.colormaps import cm
-        surf_count=0 #FIXME
+        if ds_name is None:
+            from PYME.experimental._triangle_mesh import TriangleMesh
+            mesh_ds = [k for k, d in self.pipeline.dataSources.items() if isinstance(d, TriangleMesh)]
+            ds_name = mesh_ds[0]
```

Comment:
```suggestion
            ds_name = mesh_ds[-1]
```

It's quite likely that you will want to add the most recently created mesh, rather than the first one you created (this should work on python>3.7, as dicts are ordered by default).